src/Plugin/BulletinBase.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Drupal\govdelivery_bulletins\Plugin;
+
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Queue\QueueWorkerBase;
+
+/**
+ * Provides base functionality for the NodePublish Queue Workers.
+ */
+class BulletinBase extends QueueWorkerBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * BulletinBase constructor.
+   */
+  public function __construct() {
+
+  }
+
+  /**
+   * Processes the bulletins in the queue.
+   *
+   * @param DateTime $time
+   *   The timestamp (optional) of when the queue item is to be actionalble.
```

Comment:
I think we want to be specific in the language here that the time is the end time and that only items from before this time are processed.  $time should probably become $end_time

Reply:
Makes sense, @swirtSJW :+1: